Client/mods/deathmatch/logic/CResource.cpp

```diff
@@ -267,14 +267,15 @@ void CResource::Load()
     }
 
     // Load the no cache scripts first
-    for (std::list<SNoClientCacheScript>::iterator iter = m_NoClientCacheScriptList.begin(); iter != m_NoClientCacheScriptList.end(); ++iter)
-    {
+    for (auto& item : m_NoClientCacheScriptList) {
         DECLARE_PROFILER_SECTION(OnPreLoadNoClientCacheScript)
-        const SNoClientCacheScript& item = *iter;
+
         GetVM()->LoadScriptFromBuffer(item.buffer.GetData(), item.buffer.GetSize(), item.strFilename);
+        item.buffer.ZeroClear();
+        m_NoClientCacheScriptList.pop_front();
```

Comment:
You are invalidating every iterator and reference to the first element in the list here. This is not going to work with a range-based for loop.